### ISSUE TITLE
Python lint fails due to setuptools/distutils on Ubuntu Travis builds

### ISSUE THREAD

**Issue by vladak** [oracle/opengrok#3204]
The python lint has started failng recently on Ubuntu Travis builds:
```
[INFO] --- exec-maven-plugin:1.6.0:exec (Install python lint and flake8) @ opengrok-tools ---
Collecting pylint
  Downloading pylint-2.6.0-py3-none-any.whl (325 kB)
Collecting flake8
  Downloading flake8-3.8.3-py2.py3-none-any.whl (72 kB)
Collecting toml>=0.7.1
  Downloading toml-0.10.1-py2.py3-none-any.whl (19 kB)
Collecting mccabe<0.7,>=0.6
  Downloading mccabe-0.6.1-py2.py3-none-any.whl (8.6 kB)
Collecting astroid<=2.5,>=2.4.0
  Downloading astroid-2.4.2-py3-none-any.whl (213 kB)
Collecting isort<6,>=4.2.5
  Downloading isort-4.3.21-py2.py3-none-any.whl (42 kB)
Collecting pycodestyle<2.7.0,>=2.6.0a1
  Downloading pycodestyle-2.6.0-py2.py3-none-any.whl (41 kB)
Collecting importlib-metadata; python_version < "3.8"
  Downloading importlib_metadata-1.7.0-py2.py3-none-any.whl (31 kB)
Collecting pyflakes<2.3.0,>=2.2.0
  Downloading pyflakes-2.2.0-py2.py3-none-any.whl (66 kB)
Collecting lazy-object-proxy==1.4.*
  Downloading lazy_object_proxy-1.4.3-cp35-cp35m-manylinux1_x86_64.whl (55 kB)
Collecting wrapt~=1.11
  Downloading wrapt-1.12.1.tar.gz (27 kB)
Requirement already satisfied: six~=1.12 in ./env/lib/python3.5/site-packages/six-1.15.0-py3.5.egg (from astroid<=2.5,>=2.4.0->pylint) (1.15.0)
Collecting typed-ast<1.5,>=1.4.0; implementation_name == "cpython" and python_version < "3.8"
  Downloading typed_ast-1.4.1-cp35-cp35m-manylinux1_x86_64.whl (737 kB)
Collecting zipp>=0.5
  Using cached zipp-1.2.0-py2.py3-none-any.whl (4.8 kB)
Building wheels for collected packages: wrapt
  Building wheel for wrapt (setup.py): started
  Building wheel for wrapt (setup.py): finished with status 'done'
  Created wheel for wrapt: filename=wrapt-1.12.1-cp35-cp35m-linux_x86_64.whl size=37424 sha256=4c66f2363fa85029cecd11fc629d0586e18d287e7086edf44eb74f4ad4cfbe5e
  Stored in directory: /home/travis/.cache/pip/wheels/b8/2d/88/2fb764a7404287819251978fbfe4cacf9b4d8a1d41fce5dd97
Successfully built wrapt
Installing collected packages: toml, mccabe, lazy-object-proxy, wrapt, typed-ast, astroid, isort, pylint, pycodestyle, zipp, importlib-metadata, pyflakes, flake8
Successfully installed astroid-2.4.2 flake8-3.8.3 importlib-metadata-1.7.0 isort-4.3.21 lazy-object-proxy-1.4.3 mccabe-0.6.1 pycodestyle-2.6.0 pyflakes-2.2.0 pylint-2.6.0 toml-0.10.1 typed-ast-1.4.1 wrapt-1.12.1 zipp-1.2.0
[INFO] 
[INFO] --- exec-maven-plugin:1.6.0:exec (Python lint) @ opengrok-tools ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/runpy.py", line 184, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.5/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/travis/build/oracle/opengrok/opengrok-tools/target/env/lib/python3.5/site-packages/pylint/__main__.py", line 18, in <module>
    pylint.run_pylint()
  File "/home/travis/build/oracle/opengrok/opengrok-tools/target/env/lib/python3.5/site-packages/pylint/__init__.py", line 19, in run_pylint
    from pylint.lint import Run as PylintRun
  File "/home/travis/build/oracle/opengrok/opengrok-tools/target/env/lib/python3.5/site-packages/pylint/lint/__init__.py", line 75, in <module>
    from pylint.lint.check_parallel import check_parallel
  File "/home/travis/build/oracle/opengrok/opengrok-tools/target/env/lib/python3.5/site-packages/pylint/lint/check_parallel.py", line 7, in <module>
    from pylint import reporters
  File "/home/travis/build/oracle/opengrok/opengrok-tools/target/env/lib/python3.5/site-packages/pylint/reporters/__init__.py", line 24, in <module>
    from pylint import utils
  File "/home/travis/build/oracle/opengrok/opengrok-tools/target/env/lib/python3.5/site-packages/pylint/utils/__init__.py", line 47, in <module>
    from pylint.utils.ast_walker import ASTWalker
  File "/home/travis/build/oracle/opengrok/opengrok-tools/target/env/lib/python3.5/site-packages/pylint/utils/ast_walker.py", line 6, in <module>
    from astroid import nodes
  File "/home/travis/build/oracle/opengrok/opengrok-tools/target/env/lib/python3.5/site-packages/astroid/__init__.py", line 65, in <module>
    from astroid.nodes import *
  File "/home/travis/build/oracle/opengrok/opengrok-tools/target/env/lib/python3.5/site-packages/astroid/nodes.py", line 23, in <module>
    from astroid.node_classes import (
  File "/home/travis/build/oracle/opengrok/opengrok-tools/target/env/lib/python3.5/site-packages/astroid/node_classes.py", line 42, in <module>
    from astroid import bases
  File "/home/travis/build/oracle/opengrok/opengrok-tools/target/env/lib/python3.5/site-packages/astroid/bases.py", line 36, in <module>
    MANAGER = manager.AstroidManager()
  File "/home/travis/build/oracle/opengrok/opengrok-tools/target/env/lib/python3.5/site-packages/astroid/util.py", line 26, in <lambda>
    lambda: importlib.import_module("." + module_name, "astroid")
  File "/usr/lib/python3.5/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/home/travis/build/oracle/opengrok/opengrok-tools/target/env/lib/python3.5/site-packages/astroid/manager.py", line 27, in <module>
    from astroid.interpreter._import import spec
  File "/home/travis/build/oracle/opengrok/opengrok-tools/target/env/lib/python3.5/site-packages/astroid/interpreter/_import/spec.py", line 13, in <module>
    import distutils
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 666, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 577, in module_from_spec
  File "/home/travis/build/oracle/opengrok/opengrok-tools/target/env/lib/python3.5/site-packages/_distutils_hack/__init__.py", line 82, in create_module
    return importlib.import_module('._distutils', 'setuptools')
  File "/usr/lib/python3.5/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 981, in _gcd_import
  File "<frozen importlib._bootstrap>", line 931, in _sanity_check
SystemError: Parent module 'setuptools' not loaded, cannot perform relative import
[ERROR] Command execution failed.
```
The macOS/Windows builds pass.

**Comment by vladak** [comment]
It is not clear to me yet whether this could be caused by https://github.com/pypa/setuptools/issues/2350

**Comment by vladak** [comment]
This seems plausible given that the last working build https://travis-ci.org/github/oracle/opengrok/jobs/721940602 used setuptools-49.6.0 and the next one https://travis-ci.org/github/oracle/opengrok/jobs/722691194 failed had setuptools-50.0.0.

**Comment by vladak** [comment]
Possible temporary workarounds:
  - freeze setuptools at version 49
  - use the `SETUPTOOLS_USE_DISTUTILS=stdlib` environment variable for relevant executions

**Comment by vladak** [comment]
Fix depends on #3203 to see it works.

**Comment by vladak** [comment]
Seems like this was resolved in setuptools 50.1.0
